packages/fetchai/connections/p2p_libp2p/connection.py

```diff
@@ -338,7 +343,15 @@ async def write(self, data: bytes) -> None:
         """
         if self.pipe is None:
             raise ValueError("pipe is not set.")  # pragma: nocover
-        await self.pipe.write(data)
+
+        try:
+            await self.pipe.write(data)
+        except Exception:  # pylint: disable=broad-except
+            self.logger.exception(
+                "Exception raised on message write. Try reconnect to node and write again."
+            )
+            await self.restart()
+            await self.pipe.write(data)
```

Comment:
Should we not catch the second exception here too (like in read)?

Reply:
no, just follow the original design, and this one does not return a value at all